File: src/types/session.ts
export type SessionStatus = "OPEN" | "CLOSED";

export interface Session {
  id: string;
  status: SessionStatus;
  /** En centimes */
  opening_float: number;
  opening_note: string | null;
  opened_at: string;
  closed_at: string | null;
  cloture_id: string | null;
}

export interface RapportX {
  session: Session;
  total_ventes_ttc: number;
  total_avoirs_ttc: number;
  net_ttc: number;
  nb_transactions: number;
  pay_especes: number;
  pay_cb: number;
  pay_cheque: number;
  pay_autre: number;
  /** Montants TVA collectés par taux (centimes) */
  tva_2000: number;
  tva_1000: number;
  tva_550: number;
  tva_210: number;
  tva_0: number;
  /** Bases HT correspondantes (centimes) */
  ht_2000: number;
  ht_1000: number;
  ht_550: number;
  ht_210: number;
  ht_0: number;
}

File: src-tauri/src/db/models/session.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Session {
    pub id: String,
    pub status: String,
    pub opening_float: i64,
    pub opening_note: Option<String>,
    pub opened_at: String,
    pub closed_at: Option<String>,
    pub cloture_id: Option<String>,
    pub cashier_id: Option<String>,
    pub station_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Cloture {
    pub id: String,
    pub session_id: String,
    pub sequence_no: i64,
    pub total_ventes_ttc: i64,
    pub total_avoirs_ttc: i64,
    pub total_annulations_ttc: i64,
    pub net_ttc: i64,
    pub tva_2000: i64,
    pub tva_1000: i64,
    pub tva_550: i64,
    pub tva_210: i64,
    pub tva_0: i64,
    pub pay_especes: i64,
    pub pay_cb: i64,
    pub pay_cheque: i64,
    pub pay_autre: i64,
    pub previous_hash: Option<String>,
    pub hash: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct GrandTotal {
    pub id: String,
    pub cloture_id: String,
    pub gt_ventes_ttc: i64,
    pub gt_avoirs_ttc: i64,
    pub gt_net_ttc: i64,
    pub gt_tva_2000: i64,
    pub gt_tva_1000: i64,
    pub gt_tva_550: i64,
    pub gt_tva_210: i64,
    pub gt_tva_0: i64,
    pub created_at: String,
}
